File: config/comm.go
package config

type Communication struct {
	Gotify   CommGotify
	Email    CommEmail
	Telegram CommTelegram
}

type CommTelegram struct {
	Token string
	Owner string
}

type CommEmail struct {
	SMTP struct {
		Host string
		Port int

		From string
		To   string

		UserName string
		Password string

		DefaultSubject string
	}
}

type CommGotify struct {
	BaseURL  string
	AppToken string
}

File: src/config/comm.rs
use serde::Deserialize;

#[derive(Debug, Deserialize)]
pub struct Config {
    pub email: Option<EmailConfig>,
    pub gotify: Option<GotifyConfig>,
    pub discord: Option<DiscordConfig>,
}

#[derive(Debug, Deserialize)]
pub struct EmailConfig {
    pub from: String,
    pub to: String,
    pub host: String,
    pub port: u16,
    pub username: String,
    pub password: String,
    pub default_subject: String,
}

#[derive(Debug, Deserialize)]
pub struct GotifyConfig {
    pub url: String,
    pub token: String,
}

#[derive(Debug, Deserialize)]
pub struct DiscordConfig {
    pub token: String,
    pub channel_id: u64,
}
